src/Marten.NodaTime/NodaTimeExtensions.cs

```diff
@@ -18,30 +21,63 @@ public static class NodaTimeExtensions
         /// <exception cref="NotSupportedException">Thrown if the current serializer is not supported for automatic configuration.</exception>
         public static void UseNodaTime(this StoreOptions storeOptions, bool shouldConfigureJsonSerializer = true)
         {
+            SetNodaTimeMappings();
             NpgsqlConnection.GlobalTypeMapper.UseNodaTime();
 
-            if (shouldConfigureJsonSerializer)
+            if (!shouldConfigureJsonSerializer) return;
+
+            var serializer = storeOptions.Serializer();
+
+            switch (serializer)
             {
-                var serializer = storeOptions.Serializer();
-                if(serializer is JsonNetSerializer jsonNetSerializer)
-                {
+                case JsonNetSerializer jsonNetSerializer:
                     jsonNetSerializer.Customize(s =>
                     {
                         s.ConfigureForNodaTime(DateTimeZoneProviders.Tzdb);
                     });
-                }
-                else if (serializer is SystemTextJsonSerializer systemTextJsonSerializer)
-                {
+                    break;
+                case SystemTextJsonSerializer systemTextJsonSerializer:
                     systemTextJsonSerializer.Customize(s =>
                     {
                         s.ConfigureForNodaTime(DateTimeZoneProviders.Tzdb);
                     });
-                }
-                else
-                    throw new NotSupportedException("Current serializer cannot be automatically configured for Nodatime. Set shouldConfigureJsonSerializer to false if you're using your own serializer.");
-                
-                storeOptions.Serializer(serializer);
+                    break;
+                default:
+                    throw new NotSupportedException("Current serializer cannot be automatically configured for NodaTime. Set shouldConfigureJsonSerializer to false if you're using your own serializer.");
             }
+
+            storeOptions.Serializer(serializer);
+        }
+
+        public static void SetNodaTimeMappings()
```

Comment:
Suggestion: Rename this from `SetNodaTimeMappings` to `SetNodaTimeTypeMappings` to add clarity.

Reply:
Fixed in: https://github.com/JasperFx/marten/pull/1987/commits/380b2b39d57ad6454e1fa770d20a3cb022931268